### ISSUE TITLE
Multi-label single classifier output

### ISSUE THREAD

**Issue by davemulford** [jdermody/brightwire#31]
I'm looking at the [MultiLabelSingleClassifier example](https://github.com/jdermody/brightwire/blob/master/SampleCode/MultiLabel.cs#L35-L86), and I'm struggling to understand how to print the suggested labels from the result.

What I'm trying to do is similar to choosing movie genres based on the movie summary. Below is my test code after the `IGraphTrainingEngine.Train` method is called (which is where the MultiLabel sample linked above ends.

    engine.Train(TRAINING_ITERATIONS, testData, errorMetric, null, 50);

    var executionEngine = graph.CreateEngine(engine.Graph);
    var output = executionEngine.Execute(testData);
    Console.WriteLine($"Final accuracy: {output.Average(o => o.CalculateError(errorMetric)):P2}");

    foreach (var r in output)
    {
        Console.WriteLine("=============== MINIBATCH START? ===============================");
        for (int docIndex = 0; docIndex < r.Target.Count; ++docIndex)
        {
            Console.WriteLine();
            Console.WriteLine("------- DOC START ------------");
            Console.WriteLine(documents[docIndex].Summary);
            Console.WriteLine($"OriginalCategories: {String.Join(", ", documents[docIndex].Categories)}");

            Console.Write("SuggestedCategories: ");

            var a = r.Target[docIndex].Data
                        .Select((f, fIndex) => new {
                            Value = f,
                            Index = fIndex
                        })
                        .ToList();

            Console.WriteLine(
                String.Join(", ",
                    a.Where(x => x.Value == 1.0f)
                        .Select(x => categories[x.Index])
                )
            );
        }
    }

The `output` variable is of type `IReadOnlyList<ExecutionResult>` so it seems like these are the results of each minibatch? An `ExecutionResult` object has the following properties:

- `IReadOnlyList<BrightWire.Models.FloatVector> Output` The list of output rows
- `IReadOnlyList<BrightWire.Models.FloatVector> Target` The list of target rows
- `IReadOnlyList<IReadOnlyList<BrightWire.Models.FloatVector>> Input` The list of input rows
- `IMiniBatchSequence MiniBatchSequence` The mini batch

The `Target` property seems to be what I want, but the size of that array isn't always the same batch size of 128 that I expect.

Is there a different way of pulling the results from this, or am I missing something else?

**Comment by jdermody** [comment]
The property that you want is actually **Output** on `ExecutionResult`. **Target** refers to the expected output (from training data). The size of each should be the same (the output size of the network).

Here's some code to output the results from this example. I've also added it to the example code:

```
// export the final model and execute it on the training set
var executionEngine = graph.CreateEngine(engine.Graph);
var output = executionEngine.Execute(testData);

// output the results
var rowIndex = 0;
foreach (var item in output) {
    var sb = new StringBuilder();
    foreach (var classification in item.Output.Zip(item.Target, (o, t) => (Output: o, Target: t))) {
        var columnIndex = 0;
        sb.AppendLine($"{rowIndex++}) ");
	foreach (var column in classification.Output.Data.Zip(classification.Target.Data, (o, t) => (Output: o, Target: t))) {
            var prediction = column.Output >= 0.5f ? "true" : "false";
            var actual = column.Target >= 0.5f ? "true" : "false";
            sb.AppendLine($"\t{columnIndex++}) predicted {prediction} (expected {actual})");
        }
    }
    Console.WriteLine(sb.ToString());
}
```

**Comment by davemulford** [comment]
Awesome, this example is perfect for me. Thank you!
